android/composeui/src/main/java/com/stadiamaps/ferrostar/composeui/views/TripProgressView.kt

```diff
@@ -54,92 +56,103 @@ import uniffi.ferrostar.TripProgress
  * @param distanceFormatter The formatter to use for the distance.
  * @param durationFormatter The formatter to use for the duration.
  * @param progress The progress of the trip.
+ * @param currentRoadName The current road name (if any).
+ * @param currentRoadNameView A composable that displays the current road name (when present). A
+ *   default is provided, but you may replace this with your own view.
  * @param fromDate The date to use as the reference for the estimated arrival time.
  * @param timeZone The time zone used for the estimated arrival time formatter.
  * @param onTapExit An optional callback to invoke when the exit button is tapped. If null, the exit
  *   button is not displayed.
  */
 @Composable
-fun ArrivalView(
+fun TripProgressView(
     modifier: Modifier = Modifier,
-    theme: ArrivalViewTheme = DefaultArrivalViewTheme,
+    theme: TripProgressViewTheme = DefaultTripProgressViewTheme,
     estimatedArrivalFormatter: DateTimeFormatter = EstimatedArrivalDateTimeFormatter(),
     distanceFormatter: DistanceFormatter = LocalizedDistanceFormatter(),
     durationFormatter: DurationFormatter = LocalizedDurationFormatter(),
     progress: TripProgress,
+    currentRoadName: String? = null,
+    currentRoadNameView: @Composable (String) -> Unit = {},
     fromDate: Instant = Clock.System.now(),
     timeZone: TimeZone = TimeZone.currentSystemDefault(),
     onTapExit: (() -> Unit)? = null
 ) {
   Box(modifier) {
-    Row(
-        modifier =
-            Modifier.shadow(12.dp, shape = RoundedCornerShape(50))
-                .background(color = theme.backgroundColor, shape = RoundedCornerShape(50))
-                .padding(start = 32.dp, end = 12.dp, top = 12.dp, bottom = 12.dp),
-        horizontalArrangement = Arrangement.SpaceBetween,
-        verticalAlignment = Alignment.CenterVertically) {
-          Column(
-              modifier = Modifier.weight(1f), horizontalAlignment = Alignment.CenterHorizontally) {
-                Text(
-                    text =
-                        estimatedArrivalFormatter.format(
-                            progress.estimatedArrivalTime(fromDate, timeZone)),
-                    style = theme.measurementTextStyle)
-                if (theme.style == ArrivalViewStyle.INFORMATIONAL) {
+    Column(horizontalAlignment = Alignment.CenterHorizontally) {
+      currentRoadName?.let { roadName -> currentRoadNameView(roadName) }
+      Row(
+          modifier =
+              Modifier.shadow(12.dp, shape = RoundedCornerShape(50))
+                  .background(color = theme.backgroundColor, shape = RoundedCornerShape(50))
+                  .padding(start = 12.dp, end = 12.dp, top = 12.dp, bottom = 12.dp),
```

Comment:
Actual change here from 32 to 12 start padding. This asymmetry drove me nuts; it's better now :)